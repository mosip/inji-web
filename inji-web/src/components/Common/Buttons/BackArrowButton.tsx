import React from 'react';
import BackArrowIcon from '../../../assets/BackArrowIcon.svg';

interface BackArrowButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const BackArrowButton: React.FC<BackArrowButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-back-arrow-container',
    iconTestId = 'icon-back-arrow',
    alt = 'Back Arrow'
}) => {
    return (
        <button
            type="button"
            onClick={onClick}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
