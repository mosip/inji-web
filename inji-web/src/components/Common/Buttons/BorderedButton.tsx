import React from "react";
import {renderGradientText} from "../../../utils/builder";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={props.fullWidth ? "w-full" : ""}>
    <div className={"bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"} onClick={props.onClick}>
            <div className={`py-1 px-2 rounded-lg bg-white justify-center center ${props.className}`}>
                <button
                    data-testid={props.testId}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title && renderGradientText(props.title)}
                </button>
            </div>
    </div>
  </div>
}

export type BorderedButtonProps = {
    fullWidth?: boolean | false;
    testId: string;
    onClick: (event: React.MouseEvent)=> void;
    title?: string;
    className?: string;
}
