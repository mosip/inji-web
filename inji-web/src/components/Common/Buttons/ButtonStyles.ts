export const ButtonStyles = {
    responsiveButtonWithText: {
        container: "flex items-center space-x-2 w-full",
        smallDeviceView: " w-full",
        smallDeviceButton: "sm:hidden",
        largeDeviceContainer: "hidden sm:flex items-center space-x-2",
        gradientContainer: "rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary p-[2px] transition-all duration-200 ease-in-out",
        iconButton: "flex items-center justify-center w-10 h-10 rounded-xl shadow border-none bg-[#FBF2EF] transition-all duration-200 ease-in-out",
        hoveredIconButton: "bg-gradient-to-r from-iw-primary to-iw-secondary"
    }
}