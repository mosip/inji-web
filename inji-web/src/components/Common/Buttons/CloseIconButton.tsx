import React from 'react';
import BackArrowIcon from '../../../assets/CloseIcon.svg';

interface CloseIconButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const CloseIconButton: React.FC<CloseIconButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-close-container',
    iconTestId = 'icon-close',
    alt = "Close"
}) => {
    return (
        <button
            type="button"
            onClick={(e)=>{
                e.stopPropagation()
                onClick()
            }}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
