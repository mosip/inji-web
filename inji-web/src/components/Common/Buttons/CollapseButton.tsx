import { useSelector } from "react-redux";
import { isRTL } from "../../../utils/i18n";
import { RootState } from "../../../types/redux";
import CollapseIcon from '../../../assets/CollapseIcon.svg';

type CollapseButtonProps = {
    isCollapsed: boolean;
    onClick: () => void;
    className?: string;
};

export const CollapseButton: React.FC<CollapseButtonProps> = ({isCollapsed, onClick, className}) => {
    const language = useSelector((state: RootState) => state.common.language);

    return (
        <button
            onClick={onClick}
            className={className}
        >
            <img
                src={CollapseIcon}
                alt="Collapse"
                className={`transform ${
                    isCollapsed
                        ? isRTL(language)
                            ? ''
                            : 'rotate-180'
                        : isRTL(language)
                        ? 'rotate-180'
                        : ''
                } transition-transform duration-300 min-w-[26px] min-h-[26px]`}
            />
        </button>
    );
};