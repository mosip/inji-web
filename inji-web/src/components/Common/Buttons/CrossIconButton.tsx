import React from 'react';

interface CrossIconButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
}

export const CrossIconButton: React.FC<CrossIconButtonProps> = ({
                                                                    onClick,
                                                                    btnClassName = '',
                                                                    iconClassName = '',
                                                                    btnTestId = 'btn-close-icon-container',
                                                                    iconTestId = 'icon-close'
                                                                }) => {
    return (
        <button
            type="button"
            onClick={onClick}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <svg
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
                fill="currentColor"
                aria-hidden="true"
                className={iconClassName}
                data-testid={iconTestId}
            >
                <path
                    fill-rule="evenodd"
                    d="M6.2 6.2a1 1 0 011.4 0L12 10.6l4.4-4.4a1 1 0 111.4 1.4L13.4 12l4.4 4.4a1 1 0 01-1.4 1.4L12 13.4l-4.4 4.4a1 1 0 11-1.4-1.4L10.6 12 6.2 7.6a1 1 0 010-1.4z"
                    clip-rule="evenodd"
                />
            </svg>
        </button>
    );
};
