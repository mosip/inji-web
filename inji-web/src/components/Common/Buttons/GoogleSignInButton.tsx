import { useState } from "react";
import { FcGoogle } from "react-icons/fc";

export const GoogleSignInButton:React.FC<GoogleSignInButtonProps> = (props) => {
  const [isLoading, setIsLoading] = useState(false);
  const onClickHandler = () => {
    setIsLoading(true);
    props.handleGoogleLogin();
  };

  return (
    <button
      onClick={onClickHandler}
      disabled={isLoading}
      data-testid="google-login-button"
      className="w-full bg-white flex items-center justify-center gap-2.5 break-words py-2 px-4 shadow-md border border-gray-300 rounded-xl"
    >
      <FcGoogle size={24} className="flex-shrink-0" />
      {isLoading ? props.loadingText : props.text}
    </button>
  );
};

export type GoogleSignInButtonProps = {
    handleGoogleLogin: () => void;
    loadingText: string;
    text:string
  };