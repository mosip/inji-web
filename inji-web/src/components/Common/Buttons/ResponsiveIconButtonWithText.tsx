import React, {useState} from 'react';
import {SolidButton} from "./SolidButton";
import {IconProps} from "../../../types/components";

interface ResponsiveIconButtonWithTextProps {
    icon: React.ReactElement<IconProps>;
    text: string;
    onClick: () => void;
    testId: string;
}

/**
 * On small screens:
 * [ICON] [TEXT]
 * On large screens:
 * [Separate TEXT Button on the left of it, visible on hover] [Gradient ICON BUTTON]
 */
export const ResponsiveIconButtonWithText: React.FC<ResponsiveIconButtonWithTextProps> = ({
                                                                                              icon,
                                                                                              text,
                                                                                              onClick,
                                                                                              testId
                                                                                          }) => {
    const [isHovered, setIsHovered] = useState(false);
    const enhancedIcon =React.isValidElement(icon) ? React.cloneElement(icon, {
        gradient: !isHovered,
        style: {
            color: isHovered ? "#FFFFFF" : "var(--iw-color-grayMedium)",
        }
    }): icon;

    return (
        <div
            className="flex items-center space-x-2 w-full"
        >
            {/* Small screens: icon + text */}
            <SolidButton className={"sm:hidden"} testId={`btn-${testId}`} onClick={onClick} title={text} icon={icon}
                         fullWidth/>

            {/* Large screens: icon button + text as separate element */}
            <div className="hidden sm:flex items-center space-x-2">
                {isHovered && (
                    <SolidButton testId={`btn-${testId}`} onClick={onClick} title={text}/>
                )}

                <div
                    className="rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary p-[2px] transition-all duration-200 ease-in-out">
                    <button
                        onMouseEnter={() => setIsHovered(true)}
                        onMouseLeave={() => setIsHovered(false)}
                        className={`flex items-center justify-center w-10 h-10 rounded-xl shadow border-none bg-[#FBF2EF] transition-all duration-200 ease-in-out ${isHovered ? "bg-gradient-to-r from-iw-primary to-iw-secondary " : ""}`}
                        aria-label="Gradient Icon Button"
                        onClick={onClick}
                        data-testid={`btn-${testId}-icon`}
                        onKeyDown={(event: React.KeyboardEvent<HTMLButtonElement>) => {
                            if (event.key === 'Enter' || event.key === ' ') {
                                event.preventDefault();
                                event.stopPropagation()
                                onClick();
                            }
                        }}
                    >
                        {enhancedIcon}
                    </button>
                </div>

            </div>
        </div>
    );
};
