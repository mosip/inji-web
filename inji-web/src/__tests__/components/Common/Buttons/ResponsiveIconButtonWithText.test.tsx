import React from 'react';
import {fireEvent, render, screen, within} from '@testing-library/react';
import '@testing-library/jest-dom';
import {ResponsiveIconButtonWithText} from '../../../../components/Common/Buttons/ResponsiveIconButtonWithText';

describe('ResponsiveIconButtonWithText Component', () => {
    const defaultProps = {
        testId: 'responsive-button',
        text: 'Download',
        icon: <div data-testid="test-icon">Icon</div>,
        onClick: jest.fn()
    };

    const renderButton = (props = {}) => {
        return render(
            <ResponsiveIconButtonWithText {...defaultProps} {...props} />
        );
    };

    it('matches snapshot with default props', () => {
        const {asFragment} = renderButton();

        expect(asFragment()).toMatchSnapshot();
    });

    it('calls onClick handler when button is clicked', () => {
        renderButton();

        const largeDeviceView = screen.getByTestId("non-small-device-view");
        const iconButton = within(largeDeviceView).getByTestId('btn-responsive-button');
        fireEvent.click(iconButton);

        expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
    });

    it('should render text button when hovered on icon', () => {
        renderButton();

        const largeDeviceView = screen.getByTestId("non-small-device-view");
        const iconButton = within(largeDeviceView).getByTestId('btn-responsive-button');
        fireEvent.mouseEnter(iconButton);

        expect(screen.getByTestId('btn-responsive-button-hover-view')).toBeInTheDocument();
    });

    it('should render button for both small and large devices', () => {
        renderButton();

        expect(screen.getAllByTestId('btn-responsive-button')).toHaveLength(2);

        // Check small device view
        const smallDeviceView = screen.getByTestId("small-device-view");
        expect(smallDeviceView).toBeInTheDocument();
        expect(within(smallDeviceView).getByTestId('btn-responsive-button')).toBeInTheDocument();
        expect(within(smallDeviceView).getByRole("button", {name: /Download/i})).toBeInTheDocument();

        // Check large device view
        const largeDeviceView = screen.getByTestId("non-small-device-view");
        expect(largeDeviceView).toBeInTheDocument();
        expect(within(largeDeviceView).getByTestId('btn-responsive-button')).toBeInTheDocument();
    });
});