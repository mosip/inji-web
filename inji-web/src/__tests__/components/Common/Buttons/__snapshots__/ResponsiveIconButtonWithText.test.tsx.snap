// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveIconButtonWithText Component matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2 w-full"
  >
    <span
      class=" w-full"
      data-testid="small-device-view"
    >
      <div
        class="w-full"
      >
        <button
          class="w-full bg-iw-primary text-center items-center text-iw-text font-bold text-sm px-6 py-3 rounded-lg shadow text-white mr-1 hover:shadow-lg cursor-pointer  bg-gradient-to-r from-iw-primary to-iw-secondary sm:hidden"
          data-testid="btn-responsive-button"
        >
          <div
            class="flex items-center justify-center gap-1.5"
          >
            <span>
              <div
                data-testid="test-icon"
              >
                Icon
              </div>
            </span>
            <span>
              Download
            </span>
          </div>
        </button>
      </div>
    </span>
    <div
      class="hidden sm:flex items-center space-x-2"
      data-testid="non-small-device-view"
    >
      <div
        class="rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary p-[2px] transition-all duration-200 ease-in-out"
      >
        <button
          aria-label="Gradient Icon Button"
          class="flex items-center justify-center w-10 h-10 rounded-xl shadow border-none bg-[#FBF2EF] transition-all duration-200 ease-in-out "
          data-testid="btn-responsive-button"
        >
          <div
            data-testid="test-icon"
          >
            Icon
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
